operators/src/plot/temporal_raster_mean_plot.rs

```diff
@@ -0,0 +1,438 @@
+            TemporalRasterMeanPlotQueryProcessor { raster, time_position, measurement }.boxed()
+        });
+
+        Ok(TypedPlotQueryProcessor::JsonVega(processor))
+    }
+}
+
+/// A query processor that calculates the `TemporalRasterMeanPlot` about its input.
+pub struct TemporalRasterMeanPlotQueryProcessor<P: Pixel> {
+    raster: Box<dyn RasterQueryProcessor<RasterType = P>>,
+    time_position: TemporalRasterMeanPlotTimePosition,
+    measurement: Measurement,
+}
+
+#[async_trait]
+impl<P: Pixel> PlotQueryProcessor for TemporalRasterMeanPlotQueryProcessor<P> {
+    type OutputFormat = PlotData;
+
+    fn plot_type(&self) -> &'static str {
+        TEMPORAL_RASTER_MEAN_PLOT_NAME
+    }
+
+    async fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<Self::OutputFormat> {
+        let means =
+            Self::calculate_means(self.raster.query(query, ctx), self.time_position).await?;
+
+        let plot = Self::generate_plot(means, self.measurement.clone())?;
+
+        let plot_data = plot.to_vega_embeddable(false)?;
+
+        Ok(plot_data)
+    }
+}
+
+impl<P: Pixel> TemporalRasterMeanPlotQueryProcessor<P> {
+    async fn calculate_means(
+        mut tile_stream: BoxStream<'_, Result<RasterTile2D<P>>>,
+        position: TemporalRasterMeanPlotTimePosition,
+    ) -> Result<BTreeMap<TimeInstance, MeanCalculator>> {
+        let mut means: BTreeMap<TimeInstance, MeanCalculator> = BTreeMap::new();
+
+        while let Some(tile) = tile_stream.next().await {
+            let tile = tile?;
+
+            let time = Self::time_interval_projection(tile.time, position);
+
+            let mean = means.entry(time).or_default();
+            mean.add(&tile.grid_array.data, tile.grid_array.no_data_value);
+        }
+
+        Ok(means)
+    }
+
+    #[inline]
+    fn time_interval_projection(
+        time_interval: TimeInterval,
+        time_position: TemporalRasterMeanPlotTimePosition,
+    ) -> TimeInstance {
+        match time_position {
+            TemporalRasterMeanPlotTimePosition::Start => time_interval.start(),
+            TemporalRasterMeanPlotTimePosition::Center => TimeInstance::from_millis(average_floor(
+                time_interval.start().inner(),
+                time_interval.end().inner(),
+            )),
+            TemporalRasterMeanPlotTimePosition::End => time_interval.end(),
+        }
+    }
+
+    fn generate_plot(
+        means: BTreeMap<TimeInstance, MeanCalculator>,
+        measurement: Measurement,
+    ) -> Result<AreaLineChart> {
+        let mut timestamps = Vec::with_capacity(means.len());
+        let mut values = Vec::with_capacity(means.len());
+
+        for (timestamp, mean_calculator) in means {
+            timestamps.push(timestamp);
+            values.push(mean_calculator.mean());
+        }
+
+        AreaLineChart::new(timestamps, values, measurement).map_err(Into::into)
+    }
+}
+
+struct MeanCalculator {
```

Comment:
why not reuse the stuff from the Statistics operator?

Reply:
It calculates more than the mean. In the last PR you made me split up a special case where it was possible to do less work, so… :shrug:.